app/controllers/callbacks_controller.rb

```diff
@@ -2,10 +2,12 @@
 
 class CallbacksController < Devise::OmniauthCallbacksController
   def github
-    user = User.from_omniauth(request.env['omniauth.auth'])
+    user = Github::SignInService.new.call(request.env['omniauth.auth'])
```

Comment:
I would suggest something like

```ruby
user = User.from_omniauth(request.env['omniauth.auth'])
raise MissingMembershipError unless (user.organisation & WHITELISTED_ORGS),any?
```

This will allow us to get rid of the separate service, while keeping code readable (and clearly-named)